waterbox/gpgx/cinterface/cinterface.c

```diff
@@ -528,50 +542,155 @@ GPGX_EX int gpgx_init(const char* feromextension,
 	bitmap.width = 1024;
 	bitmap.height = 512;
 	bitmap.pitch = 1024 * 4;
-	bitmap.data = alloc_invisible(2 * 1024 * 1024);
-	tempsram = alloc_invisible(24 * 1024);
-	bg_pattern_cache = alloc_invisible(0x80000);
-
-	ext.md_cart.rom = alloc_plain(32 * 1024 * 1024);
-	SZHVC_add = alloc_sealed(131072);
-	SZHVC_sub = alloc_sealed(131072);
-	ym2612_lfo_pm_table = alloc_sealed(131072);
-	vdp_bp_lut = alloc_sealed(262144);
-	vdp_lut = alloc_sealed(6 * sizeof(*vdp_lut));
-	for (int i = 0; i < 6; i++)
-		vdp_lut[i] = alloc_sealed(65536);
+	bitmap.data = alloc_plain(2 * 1024 * 1024);
+	tempsram = alloc_plain(24 * 1024);
+
+    // Initializing ram deepfreeze list
+#ifdef USE_RAM_DEEPFREEZE
+	deepfreeze_list_size = 0;
+#endif
+
+	/**
+	 * Allocating large buffers
+	 */
+
+	// cart_hw/areplay.h
+
+	action_replay.ram = alloc_plain(sizeof(uint8) * 0x10000);

```

Comment:
What's the point of all these alloc_plain calls? These seem to all make it so there's more unneeded changes in the submodule.

To be clear, alloc_plain is mostly useless, it came at a time before `malloc` actually worked. `alloc_sealed` similarly is less useful nowadays, due to dirty detection making it not useful for anything not actually changed during the core's lifetime (init time doesn't count here, since Seal() occurs after that).

Reply:
Alright, so I will revert the stack->heap conversions back to the core originals. I thought this transformation was necessary from looking at the current port, but it seems not anymore.

Reply:
To be clear, gpgx was the first waterboxed core, so it contains various practices which are antiqued by the modern waterbox setup.

Also variables in the global scope are *not* in the stack (*local* variables, i.e. variables local to the function they are declared within, are usually what's referred to for variables in the stack)